### PR TITLE
[#9] [UI] As a user, I can see the empty Dashboard page

### LINKED ISSUE #9: [UI] As a user, I can see the empty Dashboard page
## Why
When the user sign in, he should see the Dashboard page. If there is no surveys, the empty page should show.

## Acceptance Criteria
- Create an empty Dashboard page with the below components:
  - The app bar with the Nimble logo on the right, and the user's avatar on the left.
  - Current datetime.
  - A large smile emoji and the text `You’ve completed all the surveys.\nTake a moment.`
[Design](https://www.figma.com/file/i1AI0IzCWuJApX5CK0DmiW/Challenge---Web-App?type=design&node-id=70-869&t=9t3BfBxlZtL53kmn-0)
image

### PR DESCRIPTION
**Pull request by manh-t**
Close #9 

## What happened 👀

- Create the Header component for the Dashboard including Nimble logo, datetime, the circle avatar.
- Create the DashboardEmpty component.

## Insight 📝

N/A

## Proof Of Work 📹

image

### CODE REVIEW

**Comment by manh-t** [comment]
🚀 Deployed on https://64b769c92ea6a645c491ec87--manh-react-survey.netlify.app

**Comment by manh-t** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Coverage is a little low, take a look</td>
    </tr>
  </tbody>
</table>

## Coverage in All Files
File | Line Coverage | Statement Coverage | Function Coverage | Branch Coverage
---- | ------------: | -----------------: | ----------------: | --------------:
[src/../v1/requestManager.ts](../blob/bcd7fc3/src/lib/request/v1/requestManager.ts) | (9/18) 50% | (9/18) 50% | (4/6) 67% | (1/3) 33%
[src/../Dashboard/index.tsx](../blob/bcd7fc3/src/screens/Dashboard/index.tsx) | (1/2) 50% | (1/2) 50% | (0/1) 0% | (0/0) NaN%
[src/config/index.ts](../blob/bcd7fc3/src/config/index.ts) | (5/9) 56% | (5/9) 56% | (0/3) 0% | (0/4) 0%
[src/tests/error.ts](../blob/bcd7fc3/src/tests/error.ts) | (2/3) 67% | (2/3) 67% | (1/2) 50% | (1/3) 33%
[src/screens/SignIn/index.tsx](../blob/bcd7fc3/src/screens/SignIn/index.tsx) | (12/16) 75% | (12/16) 75% | (3/6) 50% | (6/6) 100%
[src/helpers/authentication.ts](../blob/bcd7fc3/src/helpers/authentication.ts) | (10/11) 91% | (10/11) 91% | (3/3) 100% | (1/2) 50%
[src/App.tsx](../blob/bcd7fc3/src/App.tsx) | (3/3) 100% | (3/3) 100% | (1/1) 100% | (0/0) NaN%
[src/index.tsx](../blob/bcd7fc3/src/index.tsx) | (3/3) 100% | (3/3) 100% | (0/0) NaN% | (0/0) NaN%
[src/../Authentication/index.ts](../blob/bcd7fc3/src/adapters/Authentication/index.ts) | (2/2) 100% | (2/2) 100% | (1/1) 100% | (0/0) NaN%
[src/adapters/Base/index.ts](../blob/bcd7fc3/src/adapters/Base/index.ts) | (7/7) 100% | (7/7) 100% | (2/2) 100% | (2/2) 100%
[src/components/Alert/index.tsx](../blob/bcd7fc3/src/components/Alert/index.tsx) | (3/3) 100% | (3/3) 100% | (2/2) 100% | (0/0) NaN%
[src/../Empty/index.tsx](../blob/bcd7fc3/src/components/Dashboard/Empty/index.tsx) | (2/2) 100% | (2/2) 100% | (1/1) 100% | (0/0) NaN%
[src/../Header/index.tsx](../blob/bcd7fc3/src/components/Dashboard/Header/index.tsx) | (2/2) 100% | (2/2) 100% | (1/1) 100% | (0/0) NaN%
[src/../index.tsx](../blob/bcd7fc3/src/components/ElevatedButton/index.tsx) | (3/3) 100% | (3/3) 100% | (1/1) 100% | (0/0) NaN%
[src/../LoadingDialog/index.tsx](../blob/bcd7fc3/src/components/LoadingDialog/index.tsx) | (2/2) 100% | (2/2) 100% | (1/1) 100% | (0/0) NaN%
Other (12 more) | (59/59) 100% | (59/59) 100% | (21/21) 100% | (12/12) 100%
Total | (125/145) 86% | (125/145) 86% | (42/52) 81% | (23/32) 72%

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 3ca77cbbf4f953f4cc3b23976c917a869c12cc1c
</p>

**Comment by tyrro** [comment]
We need a rebase here @manh-t!

**Comment by manh-t** [comment]
> We need a rebase here @manh-t!

Thanks, rebased 🙏

File: src/components/Dashboard/Header/index.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  daysAgo: string;
+  profileUrl: string;
+  children: React.ReactNode;
+  'data-test-id'?: string;
+}
+const DashboardHeader = ({ dateTime, daysAgo, profileUrl, children, ...attributes }: DashboardHeaderProps): JSX.Element => {
+  return (
+    <div className="flex flex-col min-h-screen" {...attributes}>
+      <div className="flex flex-row justify-between pt-8">
+        <NimbleLogoWhite />
```
Comment (hoangmirs) [inline src/components/Dashboard/Header/index.tsx:16]: The header is often an actionable link to return home, should we add a parent div for it?
And I think we might need to decrease the size a bit on the mobile screen 🤔

File: src/components/Dashboard/Header/index.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+interface DashboardHeaderProps {
+  dateTime: string;
+  daysAgo: string;
+  profileUrl: string;
+  children: React.ReactNode;
+  'data-test-id'?: string;
+}
+const DashboardHeader = ({ dateTime, daysAgo, profileUrl, children, ...attributes }: DashboardHeaderProps): JSX.Element => {
+  return (
+    <div className="flex flex-col min-h-screen" {...attributes}>
```
Comment (hoangmirs) [inline src/components/Dashboard/Header/index.tsx]: We should use the [semantic elements](https://www.w3schools.com/html/html5_semantic_elements.asp)
So please note that to use later 🙏

File: src/components/Dashboard/Header/index.tsx (hunk 1 above)
Comment (manh-t) [reply src/components/Dashboard/Header/index.tsx:16]: @hoangmirs currently there is no design for mobile app and there is no requirements for handle responsive UI so please ignore it 😄 

> The header is often an actionable link to return home, should we add a parent div for it?

This logo is only appeared in home so I think we don't need to handle this action 🙏

File: src/components/Dashboard/Header/index.tsx (hunk 2 above)
Comment (manh-t) [reply src/components/Dashboard/Header/index.tsx]: Thanks a lot 😍

File: src/components/Dashboard/Empty/index.test.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+
+import { render, screen } from '@testing-library/react';
+
+import DashboardEmpty from '.';
+
+describe('DashboardEmpty', () => {
+  const dataTestId = 'dashboard-empty';
+  it('renders DashboardEmpty and its components', () => {
+    render(<DashboardEmpty data-test-id={dataTestId} />);
```
Comment (hoangmirs) [inline src/components/Dashboard/Empty/index.test.tsx]: As you just use this const once, could you move it into the `it()`?
```suggestion
  it('renders DashboardEmpty and its components', () => {
  	const dataTestId = 'dashboard-empty';
    render(<DashboardEmpty data-test-id={dataTestId} />);
```

**Review by hoangmirs — APPROVED** [review]

File: src/components/Dashboard/Empty/index.test.tsx (hunk 3 above)
Comment (manh-t) [reply src/components/Dashboard/Empty/index.test.tsx]: @hoangmirs updated, thanks 🙆‍♂️

File: src/components/Dashboard/Empty/index.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+
+interface DashboardEmptyProps {
+  'data-test-id'?: string;
+}
+
+const DashboardEmpty = ({ ...attributes }: DashboardEmptyProps): JSX.Element => {
```
Comment (nvminhtue) [inline src/components/Dashboard/Empty/index.tsx]:
```suggestion
const DashboardEmpty = (attributes: DashboardEmptyProps): JSX.Element => {
```

File: src/components/Dashboard/Header/index.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import { ReactComponent as NimbleLogoWhite } from 'assets/images/icons/nimble-logo-white.svg';
+
+interface DashboardHeaderProps {
+  dateTime: string;
+  daysAgo: string;
+  profileUrl: string;
+  children: React.ReactNode;
+  'data-test-id'?: string;
+}
+const DashboardHeader = ({ dateTime, daysAgo, profileUrl, children, ...attributes }: DashboardHeaderProps): JSX.Element => {
```
Comment (nvminhtue) [inline src/components/Dashboard/Header/index.tsx]:
```suggestion
const DashboardHeader = ({ dateTime, daysAgo, profileUrl, children, ...rest }: DashboardHeaderProps): JSX.Element => {
```
we usually use `rest` term

File: src/components/Dashboard/Header/index.tsx [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+      </div>
+      <div className="flex flex-row justify-between">
+        <div className="w-1/5"></div>
+        <div className="flex flex-col text-white mt-10 flex-1">
+          <p className="text-x-small font-extrabold">{dateTime}</p>
+          <p className="text-x-large font-extrabold mt-1">{daysAgo}</p>
+        </div>
+        <div className="w-1/5"></div>
+      </div>
+      <div className="flex flex-row justify-between mt-8 grow">
```
Comment (nvminhtue) [inline src/components/Dashboard/Header/index.tsx]:
```suggestion
      <div className="flex justify-between mt-8 grow">
```

File: src/components/Dashboard/Header/index.tsx [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  'data-test-id'?: string;
+}
+const DashboardHeader = ({ dateTime, daysAgo, profileUrl, children, ...attributes }: DashboardHeaderProps): JSX.Element => {
+  return (
+    <header className="flex flex-col min-h-screen" {...attributes}>
+      <div className="flex flex-row justify-between pt-8">
+        <NimbleLogoWhite />
+        <img className="w-[36px] h-[36px] rounded-full" src={profileUrl} alt="user avatar" />
+      </div>
+      <div className="flex flex-row justify-between">
```
Comment (nvminhtue) [inline src/components/Dashboard/Header/index.tsx]:
```suggestion
      <div className="flex justify-between">
```

File: src/components/Dashboard/Header/index.tsx [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  dateTime: string;
+  daysAgo: string;
+  profileUrl: string;
+  children: React.ReactNode;
+  'data-test-id'?: string;
+}
+const DashboardHeader = ({ dateTime, daysAgo, profileUrl, children, ...attributes }: DashboardHeaderProps): JSX.Element => {
+  return (
+    <header className="flex flex-col min-h-screen" {...attributes}>
+      <div className="flex flex-row justify-between pt-8">
```
Comment (nvminhtue) [inline src/components/Dashboard/Header/index.tsx]:
```suggestion
      <div className="flex justify-between pt-8">
```

File: src/screens/Dashboard/index.tsx [hunk 9]
```diff
@@ -1,20 +1,16 @@
...
-    const userToken = getToken();
-    if (userToken) {
-      setToken(userToken.accessToken);
-    }
-  }, []);
   return (
-    <div>
-      <p>Welcome to your Dashboard {token}</p>
+    <div className="bg-black w-full h-full pl-8 pr-8">
+      <DashboardHeader dateTime="Monday, JUNE 15" daysAgo="Today" profileUrl="https://i.pravatar.cc/150?img=3">
```
Comment (nvminhtue) [inline src/screens/Dashboard/index.tsx:9]: any reason you hard-coding in the `dateTime`? can it be today?

Comment (manh-t) [reply src/screens/Dashboard/index.tsx:9]: @nvminhtue the real data will come from the integrate task so I hardcode the data temporarily for UI task 🙏

**Comment by manh-t** [comment]
@nvminhtue fixed them all 🙋

**Review by nvminhtue — APPROVED** [review]

**Pull request merged**